uint32 id
uint32 num_neigh
float32 x
float32 y
uint32[] id_neigh
uint32[] cost